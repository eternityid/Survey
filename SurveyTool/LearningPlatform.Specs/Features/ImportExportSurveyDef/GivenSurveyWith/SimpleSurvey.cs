using LearningPlatform.Domain.Common;
using LearningPlatform.Domain.SurveyDesign;
using TechTalk.SpecFlow;

namespace LearningPlatform.Specs.Features.ImportExportSurveyDef.GivenSurveyWith
{
    [Binding]
    public class SimpleSurvey
    {
        private readonly SurveyContext _surveyContext;

        public SimpleSurvey(SurveyContext surveyContext)
        {
            _surveyContext = surveyContext;
        }

        [Given(@"I have a simple survey for export")]
        public void GivenIHaveASimpleSurvey()
        {
            var create = ServiceLocator.Resolve<SurveyDesign.Factory>().Invoke();
            Survey survey = create.Survey(
                create.Folder("topFolder",
                    create.Page(
                        create.SingleSelectionQuestion("gender", "", "", q => q.OrderType = OrderType.InOrder,
                            create.Option("1", text: "male"),
                            create.Option("2", text: "female"))),
                    create.Page(
                        create.NumericQuestion("age")),
                    create.Page(
                        create.Information("information")),
                    create.Page(p=>p.NavigationButtonSettings = NavigationButtonSettings.None,
                        create.Information("Completed"))
                    ));
            survey.SurveySettings.SurveyTitle = "New Survey";

            _surveyContext.AddSurvey(survey);
        }
    }
}