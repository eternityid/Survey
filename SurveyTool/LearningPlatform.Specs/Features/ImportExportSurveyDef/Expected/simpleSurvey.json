{
  "topFolder": {
    "alias": "topFolder",
    "childNodes": [
      {
        "type": "PageDefinition",
        "questionDefinitions": [
          {
            "type": "SingleSelectionQuestionDefinition",
            "optionList": {
              "options": [
                {
                  "text": {
                    "items": [
                      {
                        "language": "en",
                        "text": "male"
                      }
                    ]
                  },
                  "alias": "1",
                  "position": 1,
                  "optionsMask": {}
                },
                {
                  "text": {
                    "items": [
                      {
                        "language": "en",
                        "text": "female"
                      }
                    ]
                  },
                  "alias": "2",
                  "position": 2,
                  "optionsMask": {}
                }
              ],
              "optionGroups": []
            },
            "optionsMask": {},
            "alias": "gender",
            "title": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "description": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "validations": []
          }
        ],
        "skipCommands": [],
        "title": {
          "items": [
            {
              "language": "en",
              "text": "Page 1"
            }
          ]
        },
        "description": {
          "items": [
            {
              "language": "en",
              "text": ""
            }
          ]
        },
        "alias": "Page1",
        "childNodes": []
      },
      {
        "type": "PageDefinition",
        "questionDefinitions": [
          {
            "type": "NumericQuestionDefinition",
            "alias": "age",
            "title": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "description": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "validations": []
          }
        ],
        "skipCommands": [],
        "title": {
          "items": [
            {
              "language": "en",
              "text": "Page 2"
            }
          ]
        },
        "description": {
          "items": [
            {
              "language": "en",
              "text": ""
            }
          ]
        },
        "alias": "Page2",
        "childNodes": []
      },
      {
        "type": "PageDefinition",
        "questionDefinitions": [
          {
            "type": "InformationDefinition",
            "alias": "information",
            "title": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "description": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "validations": []
          }
        ],
        "skipCommands": [],
        "title": {
          "items": [
            {
              "language": "en",
              "text": "Page 3"
            }
          ]
        },
        "description": {
          "items": [
            {
              "language": "en",
              "text": ""
            }
          ]
        },
        "alias": "Page3",
        "childNodes": []
      },
      {
        "type": "PageDefinition",
        "questionDefinitions": [
          {
            "type": "InformationDefinition",
            "alias": "Completed",
            "title": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "description": {
              "items": [
                {
                  "language": "en",
                  "text": ""
                }
              ]
            },
            "validations": []
          }
        ],
        "skipCommands": [],
        "navigationButtonSettings": "None",
        "title": {
          "items": [
            {
              "language": "en",
              "text": "Page 4"
            }
          ]
        },
        "description": {
          "items": [
            {
              "language": "en",
              "text": ""
            }
          ]
        },
        "alias": "Page4",
        "childNodes": []
      }
    ]
  },
  "surveySettings": {
    "surveyTitle": "New Survey",
    "enableBackButton": true,
    "keyboardSupport": true,
    "languages": [
      "en"
    ],
    "defaultLanguage": "en",
    "displayProgressBar": true,
    "displayOneQuestionOnScreen": true
  },
  "createdDate": "N/A",
  "modifiedDate": "N/A",
  "lastPublishedDate": "N/A"
}